### DISCUSSION TITLE
Serve local m3u8 file using netcat

### DISCUSSION THREAD

**Discussion by tathastu871** [streamlink/streamlink#5148] [Q&A] [answered]
### Description

Trying to  play locl m3u8 file inside termux using netcat server
{ echo -ne "HTTP/1.0 200 OK\r\nContent-Length: $(wc -c <1.m3u8)\r\n\r\n"; cat 1.m3u8; } | nc -l 8080

```am start -W -S -n org.videolan.vlc/org.videolan.vlc.gui.video.VideoPlayerActivity -a android.intent.action.VIEW -d http://127.0.0.1:8080/```
Above Works

But Below not working
```
streamlink  --player=bash -a '-c '\''am start -W -S -n org.videolan.vlc/org.videolan.vlc.gui.video.VideoPlayerActivity -a android.intent.action.VIEW  -d '\''{filename}'\'' ; read -n 1'\''' "http://127.0.0.1:8080/1.m3u8" best```


It says location cannot be played.
tried hls://http://127.0.0.1:8080/1.m3u8

**Comment by bastimeyer** [comment, marked as answer]
What you're trying to do here doesn't make any sense at all.

First off, you don't need Streamlink to play back HLS playlists. VLC can do that natively, so the entire Streamlink wrapper you're trying to build is useless. Just launch VLC with the address of the HLS playlist served on your local web server.

Second, Streamlink, when opening a player process (the default behavior), will write its output stream to the player's stdin pipe. In your case, this is bash, and the content of the `{filename}` variable will thus be `-`, which will not be understood, hence "location cannot be played". You would need to use something like `--player-continuous-http` or `--player-fifo` that'll write the stream output to somewhere the player can read from, but that will defeat the entire idea of your wrapper.

The `hls://` prefix is only required for direct HLS playlist URLs whose path doesn't end with `.m3u8`.

**Reply by tathastu871** [reply]
i am using {filename} variable and  --player-continous-http
i am using streamlink wrapper because it allows parallel thread downloading using --stream-segment-threads 
which helps prvent buffering.
when normally playing m3u8 in vlc. it keeps buffering. but when cached using streamlink its speed increases


Using python -m http.server 8000
and http://127.0.0.1:8000/1.m3u8
works


BUT USING NETCAT SERVER IT DOESNT WORK. 
thats my question

**Reply by gravyboat** [reply]
> BUT USING NETCAT SERVER IT DOESNT WORK.

This is not a netcat support forum so we won't be able to assist you since you stated yourself it works with Streamlink.
